source/platforms/github/GitHubAPI.ts

```diff
@@ -130,19 +130,24 @@ export class GitHubAPI {
       return parseInt(perilID)
     }
 
-    const info = await this.getUserInfo()
-    if (info.id) {
-      return info.id
-    }
-
     const useGitHubActionsID = process.env["GITHUB_WORKFLOW"]
     if (useGitHubActionsID) {
+      // Allow to customise the GitHub actions app ID for Github Enterprise
+      const gheActionsID = process.env["GHE_ACTIONS_BOT_USER_ID"]
+      if (gheActionsID) {
+        return parseInt(gheActionsID)
+      }
       // This is the user.id of the github-actions app (https://github.com/apps/github-actions)
       // that is used to comment when using danger in a GitHub Action
       // with GITHUB_TOKEN (https://help.github.com/en/actions/configuring-and-managing-workflows/authenticating-with-the-github_token)
       return 41898282
     }
 
+    const info = await this.getUserInfo()
```

Comment:
This reorganization will cause a breaking behavior change even if `GHE_ACTIONS_BOT_USER_ID` is unset.

Originally, if `this.getUserInfo` returns a logged in user then it would immediately return with `info.id`. Now, it will *first* check if `GITHUB_WORKFLOW` is set (and it’s always set on Github, I believe), and so it will always return a hardcoded ID

This would break our codebase, as we set an env-var, I believe `DANGER_GITHUB_TOKEN` or similar (not at my desk) to log in to a bot account with a PAT. This code-change would only allow comments from the GitHub-actions app-id of `41898282`.

Reply:
I see, now it makes more sense 😄 I think I found the env var you referred to - it seems to be `DANGER_GITHUB_API_TOKEN`.

I reverted the change, but also added a logic around it to ensure we call the GitHub API for the user only if the `DANGER_GITHUB_API_TOKEN` is set. Otherwise, I am not sure if we want to make the call, when there isn't a configured user. Do you think that makes sense?

Reply:
Unfortunately, the regular `GITHUB_TOKEN` might be a valid logged in user so it’s not sufficient to only focuse on `DANGER_GITHUB_API_TOKEN`, I think! — As I understand it, that’s why we call the `getUserInfo` API — so we figure out the permissions on the attached token — figure out if it’s a real user or what exactly.

Reply:
Hmm, ok, I reverted that part of my change. I still did the tests for completeness, but i don't think they are relevant to my change anymore. 

The test that check if the configured token is a user is a bit weird, but I think it makes sense as it check that we have called the API with the correct creds (the ones used when creating the GitHub instance). And ensure that it correctly returns the mocked user ID.

Let me know what you think.